### PR TITLE
[Needs review] Add section about writing custom documentation for modules

### PR DESCRIPTION
**Pull request by Xrayez**
It took me a while to figure out the workflow of writing documentation for custom modules since Godot 3.0. I'm not sure whether it's the most optimal way to do it, so check it out please. I might be missing something or making unnecessary steps, not to mention grammar mistakes!

### CODE REVIEW

File: development/cpp/custom_modules_in_cpp.rst [hunk 1]
```diff
@@ -340,6 +340,68 @@ shared module as target in the scons command:
...
+Writing documentation may seem like a boring task, but it is highly recommended
+to document your newly created module in order to make it easier for users to
+benefit from it. Not to mention that the code you've written one year ago may
+become indistinguishable from the code that was written by someone else, so be
+kind to your future self!
+
+There are several steps in order to setup custom docs for the module:
+
+1. Make a new directory in the root of the module. The directory name can be
+   anything, but we'll be using ``doc_classes`` name throughout this section.
```
Comment (mhilbrunner) [inline development/cpp/custom_modules_in_cpp.rst]: but we'll be using *the* ``doc_classes`` name

File: development/cpp/custom_modules_in_cpp.rst [hunk 2]
```diff
@@ -340,6 +340,68 @@ shared module as target in the scons command:
...
+know which documentation classes of the module must be merged, since the module
+may contain several classes. Replace ``ClassName`` with the name of the class
+you want to write documentation for. If you need docs for more than one class,
+append those as well.
+
+The ``get_doc_path()`` method is used by the build system to determine
+the location of the docs. In our case, they will be located in the ``doc_classes``
+directory.
+
+3. Run command line:
```
Comment (mhilbrunner) [inline development/cpp/custom_modules_in_cpp.rst]: "Run command", drop the "line"

File: development/cpp/custom_modules_in_cpp.rst [hunk 3]
```diff
@@ -340,6 +340,68 @@ shared module as target in the scons command:
...
+.. code-block:: console
+
+    ERROR: Can't write doc file: docs/doc/classes/@GDScript.xml
+       At: editor/doc/doc_data.cpp:956
+
+4. Get generated doc file from ``godot/doc/classes/ClassName.xml``
+
+5. Copy this file to ``doc_classes``, optionally edit it, then compile the engine.
+
+The build system will fetch the documentation files from ``doc_classes`` directory
```
Comment (mhilbrunner) [inline development/cpp/custom_modules_in_cpp.rst]: from *the* ``doc_classes`` directory

**Comment by Xrayez** [comment]
Yep, I felt it was a bit off, updated.

**Comment by mhilbrunner** [comment]
Thanks again!

**Pull request merged**